File: front/src/app/features/auth/services/auth.service.ts
import { Injectable } from '@angular/core';
import {RegisterRequest} from "../interfaces/registerRequest";
import {Observable} from "rxjs";
import {HttpClient} from "@angular/common/http";
import {LoginRequest} from "../interfaces/loginRequest";
import {LoginResponse} from "../interfaces/loginResponse";

@Injectable({
  providedIn: 'root'
})
export class AuthService {

  constructor(private http: HttpClient) {}

  public login(loginRequest: LoginRequest): Observable<LoginResponse> {
    return this.http.post<LoginResponse>('/api/auth/login', loginRequest);
  }

  public register(signupRequest: RegisterRequest): Observable<string> {
    return this.http.post<string>('/api/auth/register', signupRequest);
  }
}

File: back/src/main/java/com/openclassrooms/mddapi/service/AuthService.java
package com.openclassrooms.mddapi.service;

import com.openclassrooms.mddapi.payload.request.LoginRequest;
import com.openclassrooms.mddapi.payload.request.RegisterRequest;
import com.openclassrooms.mddapi.payload.response.JwtResponse;
/**
 * The interface Auth service.
 */
public interface AuthService {
    String register(RegisterRequest registerRequest);
    JwtResponse login(LoginRequest loginRequest);
}
